lib/rust/ensogl/core/src/debug/monitor.rs

```diff
@@ -804,22 +768,19 @@ impl Sampler for Fps {
         "Frames per second"
     }
     fn value(&self) -> f64 {
-        self.value
+        self.stats.fps()
     }
     fn check(&self) -> ValueCheck {
-        self.value_check
+        if self.stats.frame_counter() > 1 {
+            self.check_by_threshold(FPS_WARNING_THRESHOLD, FPS_ERROR_THRESHOLD)
+        } else {
+            // On the first frame, the FPS will be 0, which in (only) this case is a correct value.
+            ValueCheck::Correct
+        }
```

Comment:
After all these changes, this code is so similar to all other samplers, that instead of writing it by hand, we should imo add an option to the macro for conditional checks here and use macro to generate all the samplers instead. Please change the macro to allow passing a conditional value here instead.

Reply:
I plan to do this in https://www.pivotaltracker.com/story/show/181140499 - this sampler should then get simplified even further, making it possible to generate it with the macro without a need to add a conditional check.

Reply:
ok!